shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -555,7 +555,8 @@ private void copyFileToLocal(AlluxioURI srcPath, AlluxioURI dstPath)
     File dstFile = new File(dstPath.getPath());
     String randomSuffix =
         String.format(".%s_copyToLocal_", RandomStringUtils.randomAlphanumeric(8));
-    File tmpDst = new File(Name.HOME + randomSuffix);
+    File tmpDst = new File(PathUtils.concatPath( dstFile.getAbsolutePath(),
```

Comment:
@aaudiber  

- The latest code I commit can recognize whether the target path is a folder or a existing file or a target filename. 

- The code I wrote is not that so elegant, I think. If we rewrite the `AlluixoURI` parsing, it can be more intuitive and more readable. 

- The tests I used are ```bin/alluxio fs copyToLocal /LICENSE `pwd`/LICENSE```; `bin/alluxio fs copyToLocal /LICENSE LICENSE`; `bin/alluxio fs copyToLocal /LICENSE .` ; `bin/alluxio fs copyToLocal /LICENSE testfilename`; But will fail `bin/alluxio fs copyToLocal /LICENSE ./LICENSE`(the current parsing connot recongnize `./`)